File: go-service/models/category.go
package models

import "time"

// Category represents a product category
type Category struct {
	ID          int        `db:"Id" json:"id"`
	Name        string     `db:"Name" json:"name"`
	Description *string    `db:"Description" json:"description,omitempty"`
	IsActive    bool       `db:"IsActive" json:"isActive"`
	CreatedAt   time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt   *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

File: backend/Api/Models/Category/Category.cs
using System.Text.Json.Serialization;
using Api.Core.Models;

namespace Api.Models;

public class Category : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    [JsonIgnore]
    public virtual ICollection<Product> Products { get; set; } = new List<Product>();
}
